pkg/sdk/testint/secrets_gen_integration_test.go

```diff
@@ -0,0 +1,673 @@
+		require.NoError(t, err)
+
+		return secret, id
+	}
+
+	createSecretWithOAuthAuthorizationCodeFlow := func(t *testing.T, integrationId sdk.AccountObjectIdentifier, refreshToken, refreshTokenExpiryTime string, with func(*sdk.CreateWithOAuthAuthorizationCodeFlowSecretRequest)) (*sdk.Secret, sdk.SchemaObjectIdentifier) {
+		t.Helper()
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+		request := sdk.NewCreateWithOAuthAuthorizationCodeFlowSecretRequest(id, refreshToken, refreshTokenExpiryTime, integrationId)
+		if with != nil {
+			with(request)
+		}
+		err := client.Secrets.CreateWithOAuthAuthorizationCodeFlow(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupSecret(t, id))
+
+		secret, err := client.Secrets.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		return secret, id
+	}
+
+	createSecretWithBasicAuthentication := func(t *testing.T, username, password string, with func(*sdk.CreateWithBasicAuthenticationSecretRequest)) (*sdk.Secret, sdk.SchemaObjectIdentifier) {
+		t.Helper()
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+		request := sdk.NewCreateWithBasicAuthenticationSecretRequest(id, username, password)
+		if with != nil {
+			with(request)
+		}
+		err := client.Secrets.CreateWithBasicAuthentication(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupSecret(t, id))
+
+		secret, err := client.Secrets.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		return secret, id
+	}
+
+	createSecretWithGenericString := func(t *testing.T, secretString string, with func(options *sdk.CreateWithGenericStringSecretRequest)) (*sdk.Secret, sdk.SchemaObjectIdentifier) {
+		t.Helper()
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+		request := sdk.NewCreateWithGenericStringSecretRequest(id, secretString)
+		if with != nil {
+			with(request)
+		}
+		err := client.Secrets.CreateWithGenericString(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupSecret(t, id))
+
+		secret, err := client.Secrets.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		return secret, id
+	}
+
+	assertSecret := func(t *testing.T, s *sdk.Secret, expectedId sdk.SchemaObjectIdentifier, expectedSecretType, expectedComment string) {
+		t.Helper()
+		assert.Equal(t, expectedId.Name(), s.Name)
+		assert.Equal(t, expectedSecretType, s.SecretType)
+		assert.Equal(t, expectedComment, s.Comment)
+		assert.NotEmpty(t, s.CreatedOn)
+		assert.NotEmpty(t, s.DatabaseName)
+		assert.NotEmpty(t, s.SchemaName)
+		assert.NotEmpty(t, s.OwnerRoleType)
+		assert.NotEmpty(t, s.Owner)
+	}
+
+	type secretDetails struct {
+		Name                        string
+		Comment                     string
+		SecretType                  string
+		Username                    string
+		OauthAccessTokenExpiryTime  *time.Time
+		OauthRefreshTokenExpiryTime *time.Time
+		OauthScopes                 string
+		IntegrationName             string
+	}
+
+	assertSecretDetails := func(actual *sdk.SecretDetails, expected secretDetails) {
```

Comment:
I don't think we have asserts for DESC outputs, so this can stay as is.

Reply:
we do have e.g. `describe_database_snowflake_ext.go` so we could do something similar here (especially if the outputs for show and describe and similar, so it's easier to handcraft the assertions for the describe)